app/atomic_svc.py

```diff
@@ -0,0 +1,240 @@
+        and returning couples (phase_name, external_id)
+        '''
+        for obj in mitre_json.get('objects'):
+            if not obj.get('type') == 'attack-pattern':
+                continue
+            for e in obj.get('external_references'):
+                if not e.get('source_name') == 'mitre-attack':
+                    continue
+                external_id = e.get('external_id')
+                for kc in obj.get('kill_chain_phases'):
+                    if not kc.get('kill_chain_name') == 'mitre-attack':
+                        continue
+                    phase_name = kc.get('phase_name')
+                    yield (phase_name, external_id)
+
+    async def _populate_dict_techniques_tactics(self, entreprise_attack_path=None):
+        '''
+        Populate internal dictionary used to match techniques to corresponding tactics.
+        By default, use the file 'enterprise-attack.json' located in the Atomic Red Team repository.
+        You can use a specific version of this file by passing its path with the `entreprise_attack_path`
+        parameter.
+        '''
+        self.technique_to_tactics = defaultdict(list)
+        if not entreprise_attack_path:
+            entreprise_attack_path = os.path.join(self.repo_dir, 'atomic_red_team', 'enterprise-attack.json')
+
+        with open(entreprise_attack_path, 'r') as f:
+            mitre_json = json.load(f)
+
+        for phase_name, external_id in self._gen_single_match_tactic_technique(mitre_json):
+            self.technique_to_tactics[external_id].append(phase_name)
+
+    def _handle_attachment(self, attachment_path):
+        # attachment_path must be a POSIX path
+        payload_name = os.path.basename(attachment_path)
+        # to avoid collisions between payloads with the same name
+        with open(attachment_path, 'rb') as f:
+            h = hashlib.md5(f.read()).hexdigest()
+        payload_name = h[:6] + '_' + payload_name
+        shutil.copyfile(attachment_path, os.path.join(self.payloads_dir, payload_name), follow_symlinks=False)
+        return payload_name
+
+    @staticmethod
+    def _normalize_path(path, platform):
+        if platform == PLATFORMS['windows']:
+            return path.replace('\\', '/')
+        else:
+            return path
+
+    def _catch_path_to_atomics_folder(self, string, platform):
+        '''
+        Catch a path to the the atomics/ folder, and handle it in the best way
+        possible. If needed, will import a payload.
+        '''
+        regex = re.compile(r'\$?PathToAtomicsFolder((?:/[^/ \n]+)+|(?:\\[^\\ \n]+)+)')
+        payloads = []
+        if regex.search(string):
+            fullpath, path = regex.search(string).group(0, 1)
+            path = self._normalize_path(path, platform)
+
+            # take path from index 1, as it starts with /
+            path = os.path.join(self.repo_dir, 'atomics', path[1:])
+
+            # TODO handle folders
```

Comment:
remove all the TODOs in the module. it's good you're noticing the things that are left out of this iteration - but one thing I've noticed in years of software: TODOs rarely get accomplished and end up causing confusion later down the road, as the code shifts around them. instead, update the README with human-readable text about the things left out / concessions we made in this version.

Reply:
fixed in 5b66cb5, moved in README